src/runtime/HalideBuffer.h

```diff
@@ -2235,26 +2239,30 @@ class Buffer {
             }
         }
 
-        return innermost_strides_are_one;
+        return {d, innermost_strides_are_one};
```

Comment:
this d can be after d-- on 2226, is this expected?

Reply:
Yes, this is deliberate; if we 'flatten' multiple dimensions into one, we want to know the new, smaller number of dimensions.